src/Core/Domain/Country/Command/AddCountryCommand.php

```diff
@@ -0,0 +1,249 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Country\Command;
+
+use PrestaShop\PrestaShop\Core\Domain\Country\ValueObject\CountryZipCodeFormat;
+use Tools;
+
+/**
+ * Adds new zone with provided data.
+ */
+class AddCountryCommand
+{
+    /**
+     * @var string[]
+     */
+    protected $localizedNames;
+
+    /**
+     * @var string
+     */
+    protected $isoCode;
+
+    /**
+     * @var int
+     */
+    protected $callPrefix;
+
+    /**
+     * @var int
+     */
+    protected $defaultCurrency = 0;
+
+    /**
+     * @var int|null
+     */
+    protected $zoneId;
+
+    /**
+     * @var bool
+     */
+    protected $needZipCode = false;
+
+    /**
+     * @var ?CountryZipCodeFormat
+     */
+    protected $zipCodeFormat;
+
+    /**
+     * @var string
+     */
+    protected $addressFormat = '';
+
+    /**
+     * @var bool
+     */
+    protected $enabled = false;
+
+    /**
+     * @var bool
+     */
+    protected $containsStates = false;
+
+    /**
+     * @var bool
+     */
+    protected $needIdNumber = false;
+
+    /**
+     * @var bool
+     */
+    protected $displayTaxLabel = false;
+
+    /**
+     * @var int[]
+     */
+    protected $shopAssociation = [];
+
+    public function __construct(
```

Comment:
Quick question, why don't you use all the fields in the constructor? Usually the create commands are used to create the object fully, and the update command however can be used for partial update so that have many nullable fields allowing us to detect which field is really intended to be modified

I guess you only kept the mandatory fields in the constructor but when I look at the handler it seems like you still use most of the getters (so with their default values) maybe it would be relevant to have all the parameters definable in the constructor, even if they have optional values, so that you can get rid of the setters.

The add command would then become immutable which is nice because the object creation is enough to fully define the expected result (no risk to modify its state/content later)